Add GetImages method on camera (#1659)
viamrobotics/docs: docs/components/camera/_index.md

{{% /tab %}}
{{< /tabs >}}

### GetImages

Get simultaneous images from different sensors, along with associated metadata.

> This should not be used for getting a time series of images from the same sensor.

Reviewer: Is the docs more for users, or developers, or both? Because this statement of about how the method should not be used for getting a time series of images is much more a comment for a developer implementing GetImages. For a user, I think the phrase would be like "the multiple images returned from GetImages will never represent a time series of images"

Author: That sounds good. I was leaning on the python and go docs

{{< tabs >}}
{{% tab name="Python" %}}

**Parameters:**

- None.

**Returns:**

- [(List[NamedImage])](https://python.viam.dev/autoapi/viam/media/video/index.html#viam.media.video.NamedImage): The list of images returned from the camera system.
- [(ResponseMetadata)](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.ResponseMetadata): The metadata timestamp with this response.

```python {class="line-numbers linkable-line-numbers"}
my_camera = Camera.from_robot(robot=robot, name="my_camera")

> frame = await my_cam.get_images()

Reviewer: there are two returns, so I think this example should be something like 
```
images, metadata = await my_cam.get_images()
img0 = images[0].image
timestamp = metadata.captured_at
```

> frame = await my_cam.get_images()

Author: ```suggestion
images, metadata = await my_cam.get_images()
img0 = images[0].image
timestamp = metadata.captured_at
```

```

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/camera/index.html#viam.components.camera.Camera.get_images).

{{% /tab %}}
{{% tab name="Go" %}}

**Parameters:**

- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.

**Returns:**

- [([]image.Image)](https://pkg.go.dev/image#Image): The list of images returned from the camera system.
- [(time.Time)](https://pkg.go.dev/time#Time): The metadata timestamp with this response.

> - [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.

Reviewer: These have recently been updated in Go

```
- [([]camera.NamedImage)](https://github.com/viamrobotics/rdk/blob/5486114bf2bb29975c328242a86239216b3f73b2/components/camera/camera.go#L66): The list of images returned from the camera system, with the name of the sensor associated with the image.
- [(resource.ResponseMetadata)](https://github.com/viamrobotics/rdk/blob/5486114bf2bb29975c328242a86239216b3f73b2/resource/response_metadata.go#L10): The metadata, which holds timestamp of when the data was captured.
- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
```

Author: Ok - let's wait until those updates are in the go docs then

```go {class="line-numbers linkable-line-numbers"}
myCamera, err := camera.FromRobot(robot, "my_camera")

> images, err := myCamera.Images(context.Background())

Reviewer: the returns should be 

```
images, metadata, err := myCamera.Images(context.Background())
```

{{% /tab %}}
{{< /tabs >}}

### GetImages

> Get simultaneous images from different sensors, along with associated metadata.

Author: ```suggestion
Get simultaneous images from different imagers, along with associated metadata.
```

> This should not be used for getting a time series of images from the same sensor.

Author: ```suggestion
The multiple images returned from GetImages do not represent a time series of images.
```

- [(ResponseMetadata)](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.ResponseMetadata): The metadata timestamp with this response.

```python {class="line-numbers linkable-line-numbers"}
my_camera = Camera.from_robot(robot=robot, name="my_camera")

images, metadata = await my_cam.get_images()
img0 = images[0].image
timestamp = metadata.captured_at
```

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/camera/index.html#viam.components.camera.Camera.get_images).

{{% /tab %}}
{{% tab name="Go" %}}

**Parameters:**

- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.

**Returns:**

- [([]image.Image)](https://pkg.go.dev/image#Image): The list of images returned from the camera system.
- [(time.Time)](https://pkg.go.dev/time#Time): The metadata timestamp with this response.
- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.

```go {class="line-numbers linkable-line-numbers"}
myCamera, err := camera.FromRobot(robot, "my_camera")

> images, err := myCamera.Images(context.Background())

Author: ```suggestion
images, metadata, err := myCamera.Images(context.Background())
```
